addons/mass_mailing_sms/models/mailing_list.py

```diff
@@ -63,3 +63,106 @@ def _mailing_get_opt_out_list_sms(self, mailing):
         opt_out_contacts = subscriptions.filtered(lambda sub: sub.opt_out).mapped('contact_id')
         opt_in_contacts = subscriptions.filtered(lambda sub: not sub.opt_out).mapped('contact_id')
         return list(set(c.id for c in opt_out_contacts if c not in opt_in_contacts))
+
+    def action_merge(self, src_lists):
```

Comment:
I've been messing around with the string manipulation, but I've encountered a problem. 
When I try to add a string to the query (like the condition that changes if mobile is installed or not) it automatically adds single quotes around it.

For example

```
...
self.env.cr.execute(""" SELECT * FROM mail_contact
                        WHERE %s;""", (where_sql))
...
```
Will be outputted as
```
SELECT * FROM mail_contact
WHERE 'contact.email_normalized NOT IN (select email from mail_blacklist where active = TRUE)
OR contact.email_normalized NOT IN (select email from mail_blacklist where active = TRUE)';
```

I thought I fixed it by changing to
```
...
self.env.cr.execute(""" SELECT * FROM mail_contact
                        WHERE %s;""" % (where_sql))
...
```
But the runbot doesn't like this solution as it may be the target of SQL injection.

Any idea how I can fix that?

Thanks